truss/templates/server/common/truss_server.py

```diff
@@ -154,6 +156,31 @@ def post(self, name: str):
         self.write(final_response)
 
 
+class LivenessAndLoadHandler(TrussHTTPHandler):
+
+    _load_thread: Thread = None
+
+    @staticmethod
+    def _exec_load(model):
+        model.load()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def get(self):
+        for name in self.models:
+            model = self.models[name]
+
+            # if the model is ready or loading is in flight
+            if not model.ready and not self._load_thread:
+                self._load_thread = Thread(
```

Comment:
There is only one load thread `self._load_thread` and it loads one model. Do we need a thread per model?

Reply:
This is a bit of a quick and dirty way to grab the first model in the model list, but in reality we only ever have one model registered here. Our models are hardcoded with name `model` and only one is ever added.  This could be replaced with `list(self.models.values())[0]`, but I opted for the single iterator for a semblance of safety.

Reply:
True, we only ever have one model registered right now (we've talked about bundling multiple models together for resource efficiency but we don't know when and how we will support that). I think TrussServer is built to support multiple models, we just instantiate it with one model. I think it's ok to support only one model here. I'd suggest placing an assertion at the beginning of this function, fail loudly otherwise, and remove the for loop.

Reply:
Yeah, makes sense if we want to eventually support. This rev has an assertion to protect that it only supports 1 model, which we can update in the future.